### PR TITLE
Anonymous access

### PR DESCRIPTION
**Pull request by AinaArd**

### CODE REVIEW

File: src/main/java/ru/itis/controllers/SearchController.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+    private UserService userService;
+
+    @Autowired
+    public SearchController(UserService userService) {
+        this.userService = userService;
+    }
+
+    @CrossOrigin
+    @PreAuthorize("isAnonymous()")
```
Comment (Dimanaux) [inline src/main/java/ru/itis/controllers/SearchController.java]: But can authenticated user perform search?

File: src/main/java/ru/itis/controllers/WishListController.java [hunk 2]
```diff
@@ -28,28 +28,30 @@ public WishListController(ItemService itemService, WishListService wishListServi
...
+            return WishListDto.from(wishListService.getDefaultWishList());
         }
     }
 
     @CrossOrigin
     @PostMapping("/lists/{list-id}")
     @PreAuthorize("isAuthenticated()")
     @ApiOperation("Add new item to wish list")
-    public void addNewItem(@PathVariable Long listId, ItemForm itemForm) {
+    public void addNewItem(@PathVariable("list-id") Long listId, ItemForm itemForm, @RequestHeader(name = "AUTH") String token) {
```
Comment (Dimanaux) [inline src/main/java/ru/itis/controllers/WishListController.java]: Again?

File: src/main/java/ru/itis/controllers/WishListController.java [hunk 3]
```diff
@@ -28,28 +28,30 @@ public WishListController(ItemService itemService, WishListService wishListServi
...
         itemService.addNewItem(itemForm, listId);
     }
 
     @CrossOrigin
     @DeleteMapping("/lists/{list-id}")
     @PreAuthorize("isAuthenticated()")
     @ApiOperation("Delete an item")
-    public void deleteItem(@PathVariable Long listId, @RequestParam(name = "title") String itemName) {
+    public void deleteItem(@PathVariable("list-id") Long listId, @RequestParam("title") String itemName,
```
Comment (Dimanaux) [inline src/main/java/ru/itis/controllers/WishListController.java]: Why?

File: src/main/java/ru/itis/services/WishListService.java [hunk 4]
```diff
@@ -39,13 +39,17 @@ public WishList addNewWL(String title, String token) {
 
     public void removeByTitle(String title, String token) {
         Optional<WishList> wishListCandidate = wishListRepository.findByTitle(title);
-        if(wishListCandidate.isPresent()) {
+        if (wishListCandidate.isPresent()) {
             wishListRepository.delete(wishListCandidate.get());
             Optional<Token> tokenCandidate = tokensRepository.findByValue(token);
             User currentUser = tokenCandidate.orElseThrow(IllegalArgumentException::new).getUser();
             currentUser.getWishLists().remove(wishListCandidate.get());
```
Comment (Dimanaux) [inline src/main/java/ru/itis/services/WishListService.java]: I don't think it works

File: src/main/java/ru/itis/controllers/WishListController.java [hunk 5]
```diff
@@ -28,28 +28,30 @@ public WishListController(ItemService itemService, WishListService wishListServi
     @GetMapping("/lists/{list-id}")
     @PreAuthorize("isAuthenticated()")
     @ApiOperation("Get a single wish list")
-    public WishListDto getWishList(@PathVariable Long listId) {
+    public WishListDto getWishList(@PathVariable("list-id") Long listId, @RequestHeader(name = "AUTH") String token) {
+//      TODO: if anonym --> ability to reserve an item
         Optional<WishList> wishListCandidate = wishListService.findWishListById(listId);
```
Comment (Dimanaux) [inline src/main/java/ru/itis/controllers/WishListController.java]: Looks like that you can replace it with
```java
WishListDto.from(wishListCandidate.orElse(defaultWishList))
```

Comment (Dimanaux) [inline src/main/java/ru/itis/controllers/WishListController.java]: It is very bad having Null Object and returning optional at the same time. You can just return default object.
```java
WishList wishListService.findWishListById(listId) {
    return repository.byId(listId).orElse(defaultWishList);
}
```

File: src/main/java/ru/itis/services/WishListService.java [hunk 6]
```diff
@@ -39,13 +39,17 @@ public WishList addNewWL(String title, String token) {
...
         } else {
-            throw new IllegalArgumentException("Can not find such wish list");
+            System.out.println("No such wish list");
```
Comment (Dimanaux) [inline src/main/java/ru/itis/services/WishListService.java]: System.out? Fix this plz, return 422, 404 error or something like this.

File: src/main/java/ru/itis/services/WishListService.java [hunk 7]
```diff
@@ -39,13 +39,17 @@ public WishList addNewWL(String title, String token) {
...
         }
     }
+
+    public WishList getDefaultWishList() {
```
Comment (Dimanaux) [inline src/main/java/ru/itis/services/WishListService.java]: What does this method do?

File: src/main/java/ru/itis/services/WishListService.java [hunk 8]
```diff
@@ -39,13 +39,17 @@ public WishList addNewWL(String title, String token) {
 
     public void removeByTitle(String title, String token) {
         Optional<WishList> wishListCandidate = wishListRepository.findByTitle(title);
-        if(wishListCandidate.isPresent()) {
+        if (wishListCandidate.isPresent()) {
             wishListRepository.delete(wishListCandidate.get());
             Optional<Token> tokenCandidate = tokensRepository.findByValue(token);
             User currentUser = tokenCandidate.orElseThrow(IllegalArgumentException::new).getUser();
```
Comment (Dimanaux) [inline src/main/java/ru/itis/services/WishListService.java]: What a mess. Why does Wish list service know about tokens?

File: src/main/java/ru/itis/controllers/WishListController.java (hunk 2 above)
Comment (AinaArd) [reply src/main/java/ru/itis/controllers/WishListController.java]: Works only so

File: src/main/java/ru/itis/services/WishListService.java (hunk 7 above)
Comment (AinaArd) [reply src/main/java/ru/itis/services/WishListService.java]: deleted

File: src/main/java/ru/itis/services/WishListService.java (hunk 6 above)
Comment (AinaArd) [reply src/main/java/ru/itis/services/WishListService.java]: fixed

File: src/main/java/ru/itis/controllers/WishListController.java (hunk 5 above)
Comment (AinaArd) [reply src/main/java/ru/itis/controllers/WishListController.java]: fixed

File: src/main/java/ru/itis/controllers/SearchController.java (hunk 1 above)
Comment (AinaArd) [reply src/main/java/ru/itis/controllers/SearchController.java]: fixed

**Pull request merged**